modules/recaptcha/lib/open_project/recaptcha/engine.rb

```diff
@@ -0,0 +1,44 @@
+require 'open_project/plugins'
+require 'recaptcha'
+
+module OpenProject::Recaptcha
+  class Engine < ::Rails::Engine
+    engine_name :openproject_recaptcha
+
+    include OpenProject::Plugins::ActsAsOpEngine
+
+    register 'openproject-recaptcha',
+             author_url: 'http://openproject.com',
+             settings: {
+               default: {
+                 recaptcha_type: 'disabled'
+               }
+             },
+             bundled: true do
+      menu :admin_menu,
+           :plugin_recaptcha,
+           { controller: '/recaptcha/admin', action: :show },
+           parent: :authentication,
+           caption: ->(*) { I18n.t('recaptcha.label_recaptcha') }
+    end
+
+    config.after_initialize do
+      OpenProject::Authentication::Stage.register(:recaptcha,
+                                                  nil,
+                                                  run_after_activation: true,
+                                                  active: -> {
+                                                    type = Setting.plugin_openproject_recaptcha[:recaptcha_type]
+                                                    type.present? && type.to_s != 'disabled'
```

Comment:
Can you put the allowed values for the `recaptcha_type` setting into an enum or something? I guess the Ruby way would be constants, similar to how we do it in `Principal::STATUSES`. I've seen the "magic string" `disabled` being used in three different places and it would be better if it was defined in just one place.